app/recipes/tests/test_recipes_api.py

```diff
@@ -0,0 +1,259 @@
+        self.assertEqual(response.json(), [
+            {
+                'id': sandwich_1_id,
+                'name': 'Cheese sandwich',
+                'description': 'Bread on the outside, cheese in the middle',
+                'ingredients': [
+                    {'name': 'Bread'},
+                    {'name': 'Cheese'},
+                    {'name': 'Butter'},
+                ]
+            },
+            {
+                'id': sandwich_2_id,
+                'name': 'Bread sandwich',
+                'description': '',
+                'ingredients': [{'name': 'Bread'}]
+            },
+        ])
+
+    def test_read_recipe(self):
+        """GET /recipes/{recipe_id} should return the specified recipe"""
+        recipe_id = create_sample_recipe(
+            'Cheese sandwich',
+            ['Bread', 'Cheese', 'Butter'],
+            'Bread on the outside, cheese in the middle',
+        ).id
+
+        response = self.client.get(recipe_url(recipe_id))
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), {
+            'id': recipe_id,
+            'name': 'Cheese sandwich',
+            'description': 'Bread on the outside, cheese in the middle',
+            'ingredients': [
+                {'name': 'Bread'},
+                {'name': 'Cheese'},
+                {'name': 'Butter'},
+            ]
+        })
+
+    def test_partially_update_recipe(self):
+        """PATCH /recipes/{recipe_id} should update the provided fields of the
+            specified recipe"""
+        recipe_id = create_sample_recipe(
+            'Cheese sandwich',
+            ['Bread', 'Cheese', 'Butter'],
+            'Bread on the outside, cheese in the middle',
+        ).id
+
+        payload = {
+            'description': 'The 2nd-most boring sandwich',
+            'ingredients': [
+                {'name': 'Bread'},
+                {'name': 'Cheese'},
+            ]
+        }
+        response = self.client.patch(recipe_url(recipe_id), payload,
+                                     format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), {
+            'id': recipe_id,
+            'name': 'Cheese sandwich',
+            'description': 'The 2nd-most boring sandwich',
+            'ingredients': [
+                {'name': 'Bread'},
+                {'name': 'Cheese'},
+            ]
+        })
+        updated_recipe = Recipe.objects.get(id=recipe_id)
+        self.assertEqual(updated_recipe.description,
+                         'The 2nd-most boring sandwich')
+        updated_ingredients = Ingredient.objects.filter(recipe=updated_recipe)
+        self.assertEqual(updated_ingredients.count(), 2)
+        self.assertEqual(updated_ingredients[0].name, 'Bread')
+        self.assertEqual(updated_ingredients[1].name, 'Cheese')
+
+    def test_delete_recipe(self):
```

Comment:
Is this test actually necessary? I've written zero lines of code relating to model deletion, so it seems like this is just testing DRF's built-in `ModelViewSet` deletion behaviour. And we should be safe to assume that works! 😄